alert_tools/tools.py

```diff
@@ -0,0 +1,114 @@
+import numpy as np
+import pandas as pd
+import matplotlib.pyplot as plt
+
+from astropy.time import Time
+from astropy.io import fits
+import astropy.units as u
+import aplpy
+
+def make_dataframe(packet):
+    dfc = pd.DataFrame(packet['candidate'], index=[0])
+    df_prv = pd.DataFrame(packet['prv_candidates'])
+    dflc = pd.concat([dfc,df_prv], ignore_index=True,sort=True)	
+    dflc.objectId = packet['objectId']	
+    dflc.candid = packet['candid']
+    return dflc
+	
+
+def get_dcmag(dflc, match_radius_arcsec=1.5, star_galaxy_threshold = 0.4,band=2):
+    if (dflc.loc[0,'distpsnr1'] > match_radius_arcsec) & (dflc.loc[0,'sgscore1'] < star_galaxy_threshold):
+        print('Object is not a variable star.')
+                
+    else:
+        dflc=dflc.fillna(np.nan)
+        def robust_median(x):
+            if len(x) == 0:
+                return np.nan
+            else:
+                return np.median(x[np.isfinite(x)])
+        grp = dflc.groupby(['fid','field','rcid'])
+        impute_magnr = grp['magnr'].agg(robust_median)
+        #print(impute_magnr)
+        impute_sigmagnr = grp['sigmagnr'].agg(robust_median)
+        #print(impute_sigmagnr)
+        for idx, grpi in grp:
+            w = np.isnan(grpi['magnr'])
+            w2 = grpi[w].index
+            dflc.loc[w2,'magnr'] = impute_magnr[idx]
+            dflc.loc[w2,'sigmagnr'] = impute_sigmagnr[idx]
+        dflc['sign'] = 2* (dflc['isdiffpos'] == 't') - 1
+        
+        dflc['dc_mag'] = -2.5 * np.log10(10**(-0.4*dflc['magnr']) + dflc['sign'] * 10**(-0.4*dflc['magpsf']))   #u
+        dflc['dc_sigmag'] = np.sqrt(
+            (10**(-0.4*dflc['magnr'])* dflc['sigmagnr']) **2. + 
+            (10**(-0.4*dflc['magpsf']) * dflc['sigmapsf'])**2.) / 10**(-0.4*dflc['magnr']) + dflc['sign'] * 10**(-0.4*dflc['magpsf']) #u
+        dflc['dc_mag_ulim'] = -2.5 * np.log10(10**(-0.4*dflc['magnr']) + 10**(-0.4*dflc['diffmaglim']))   #v
+        dflc['dc_mag_llim'] = -2.5 * np.log10(10**(-0.4*dflc['magnr']) - 10**(-0.4*dflc['diffmaglim']))   #v2
+        
+        def band_amplitude(dflc, band=2):
```

Comment:
Let's make band amplitude a new function and have it use dcmag if it is present.